### PR TITLE
Add changes from production server

### PR DESCRIPTION
**Pull request by dgarciabriseno**
There's a lot go through here...

### CODE REVIEW

File: docroot/docs/index.php [hunk 1]
```diff
@@ -464,7 +464,7 @@ function renderEndpoint($endpoint, $xml=null) {
                 </div>
             </div>
 <?php
-        if ( count($endpoint->parameters->parameter) > 0 ) {
+        if ( count($endpoint->parameters->parameter > 0) ) {
```
Comment (dgarciabriseno) [inline docroot/docs/index.php:467]: Is this a bug?

File: docroot/jpipjs/css/tabs.css [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (dgarciabriseno) [inline docroot/jpipjs/css/tabs.css:1]: This file can be deleted.

File: docroot/jpipjs/astronomy.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (dgarciabriseno) [inline docroot/jpipjs/astronomy.html:1]: What is this jpipjs page? It's not accessible on the production site.

Comment (wafels) [reply docroot/jpipjs/astronomy.html:1]: That's from another website - it's not ours and should not appear in our repo.

File: docroot/resources/JSON/celestial-bodies/soho/jupiter/soho_jupiter_2000-04-25_2000-05-22.json [hunk 4]
```diff
@@ -0,0 +1 @@
...: 0.9936380881171725, "distance_sun_to_body_au": 4.988199280906329, "behind_plane_of_sun": "True", "x": 24242.370076277053, "y": -372.64520125290164}, "958524580851": {"distance_observer_to_body_au": 5.973592500364647, "distance_sun_to_observer_au": 0.9936440680173972, "distance_sun_to_body_au": 4.988203484510518, "behind_plane_of_sun": "True", "x": 24297.47835856895, "y": -365.60361465252777}, "958526380838": {"distance_observer_to_body_au": 5.9735652822418945, "distance_sun_to_observer_au": 0.993650047416593, "distance_sun_to_body_au": 4.988207688326153, "behind_plane_of_sun": "True", "x": 24352.586013877353, "y": -358.5610427421659}, "958528180824": {"distance_observer_to_body_au": 5.9735379795447745, "distance_sun_to_observer_au": 0.993656027582738, "distance_sun_to_body_au": 4.988211892353224, "behind_plane_of_sun": "True", "x": 24407.692902635546, "y": -351.5174698351151}, "958529980810": {"distance_observer_to_body_au": 5.973510591783066, "distance_sun_to_observer_au": 0.9936620079798624, "distance_sun_to_body_au": 4.988216096591732, "behind_plane_of_sun": "True", "x": 24462.79897245809, "y": -344.47294142292856}, "958531780797": {"distance_observer_to_body_au": 5.9734831181
```
Comment (dgarciabriseno) [inline docroot/resources/JSON/celestial-bodies/soho/jupiter/soho_jupiter_2000-04-25_2000-05-22.json:1]: Are these resources files required? I don't see any references to "celestial-bodies" but I do see references to "celestial-objects" and it is in the .gitignore.

File: docroot/jpipjs/astronomy.html (hunk 3 above)
Comment (dgarciabriseno) [reply docroot/jpipjs/astronomy.html:1]: Ok, will remove it. That will shrink this a lot.

File: docroot/resources/JSON/celestial-objects-psp-only/soho/jupiter/soho_jupiter_1995-12-07_1996-01-01.json [hunk 5]
```diff
@@ -0,0 +1 @@
...": "True", "x": 28766.654903618957, "y": -2497.6876276712055}, "820201910139": {"distance_observer_to_body_au": 6.232683712764281, "distance_sun_to_observer_au": 0.9735083052013624, "distance_sun_to_body_au": 5.270495337002803, "behind_plane_of_sun": "True", "x": 28825.766776211454, "y": -2504.2553577814356}, "820203710112": {"distance_observer_to_body_au": 6.232629269534547, "distance_sun_to_observer_au": 0.9735075476843807, "distance_sun_to_body_au": 5.270488101100862, "behind_plane_of_sun": "True", "x": 28884.879269919715, "y": -2510.8204063627636}, "820205510085": {"distance_observer_to_body_au": 6.232574733708684, "distance_sun_to_observer_au": 0.9735067923998594, "distance_sun_to_body_au": 5.270480865127433, "behind_plane_of_sun": "True", "x": 28943.992385553644, "y": -2517.382770280082}, "820207310057": {"distance_observer_to_body_au": 6.232520105292881, "distance_sun_to_observer_au": 0.9735060393504006, "distance_sun_to_body_au": 5.270473629082518, "behind_plane_of_sun": "True", "x": 29003.106123924645, "y": -2523.942446400098}, "820209110030": {"distance_observer_to_body_au": 6.232465384293326, "distance_sun_to_observer_au": 0.9735052885386033, "distance_sun_to_body_au": 5
```
Comment (dgarciabriseno) [inline docroot/resources/JSON/celestial-objects-psp-only/soho/jupiter/soho_jupiter_1995-12-07_1996-01-01.json:1]: I don't see this celestial-objects-psp-only folder being used either.

File: docroot/statistics/bokeh/Jhv_movies/histogram.html [hunk 6]
```diff
@@ -0,0 +1,85 @@
+
```
Comment (dgarciabriseno) [inline docroot/statistics/bokeh/Jhv_movies/histogram.html:1]: What about this bokeh stuff, these smell auto-generated.

File: scripts/check_data_injection2.php [hunk 7]
```diff
@@ -0,0 +1,67 @@
+<?php
```
Comment (dgarciabriseno) [inline scripts/check_data_injection2.php:1]: There shouldn't be a check_data_injection2.php, so one of these is correct and one of these isn't.

Comment (dgarciabriseno) [reply scripts/check_data_injection2.php:1]: maybe a problem for later, though.

File: scripts/hv_stats/AIA_data_gaps_test.csv [hunk 8]
```diff
@@ -0,0 +1,32 @@
+"2015-12-28 00:00:00","2015-12-29 00:00:00"
```
Comment (dgarciabriseno) [inline scripts/hv_stats/AIA_data_gaps_test.csv:1]: I can probably remove this "test" file.

Comment (dgarciabriseno) [reply scripts/hv_stats/AIA_data_gaps_test.csv:1]: these*

File: scripts/hv_stats/Jhv_movies/histogram.html [hunk 9]
```diff
@@ -0,0 +1,85 @@
+
```
Comment (dgarciabriseno) [inline scripts/hv_stats/Jhv_movies/histogram.html:1]: Is this autogenerated? Where is it getting its data from?

File: src/Helper/Serialize.php [hunk 10]
```diff
@@ -28,7 +28,7 @@ public function __construct($subPath, $filename, $maxAgeSec=86000) {
 
         // Verify that cache directory exists
         if ( !@file_exists($this->_path) ) {
-            if ( !@mkdir($this->_path, 0775, true) ) {
+            if ( !@mkdir($this->_path, 0777, true) ) {
```
Comment (dgarciabriseno) [inline src/Helper/Serialize.php:31]: bad voodoo

File: docroot/resources/JSON/celestial-bodies/soho/jupiter/soho_jupiter_2000-04-25_2000-05-22.json (hunk 4 above)
Comment (dgarciabriseno) [reply docroot/resources/JSON/celestial-bodies/soho/jupiter/soho_jupiter_2000-04-25_2000-05-22.json]: I added these to the gitignore. If we need them later they can be added back. But for now I think this is ok.

File: docroot/docs/index.php (hunk 1 above)
Comment (wafels) [reply docroot/docs/index.php:467]: No idea if this is a bug or not... I don't know about this function.

File: docroot/jpipjs/astronomy.html (hunk 3 above)
Comment (wafels) [reply docroot/jpipjs/astronomy.html]: Yes, the entire jpipjs directory should not be part of our rep0.

File: scripts/check_data_injection.php [hunk 11]
```diff
@@ -20,6 +20,9 @@ function grab_dump($var){
 	$logBody = "////////////////////\n".date("Y-m-d H:i:s")."\n";
 	$logBody .= $output."";
 	//$strOut = array();
+	putenv("http_proxy=http://gs600-squid.ndc.nasa.gov:443");
+	putenv("https_proxy=http://gs600-squid.ndc.nasa.gov:443");
+	putenv("http_export=http://gs600-squid.ndc.nasa.gov:443");
```
Comment (wafels) [inline scripts/check_data_injection.php:25]: So all of these are specific to running at GFSC. Maybe each of these variables should be in a config file somewhere?

File: docroot/docs/index.php (hunk 1 above)
Comment (dgarciabriseno) [reply docroot/docs/index.php:467]: We'll leave it for now. The same change is made lower in this file, so it seems intentional.
Though I'm not sure what this page is supposed to be. It's not working right now: https://api.helioviewer.org/docs 
The real docs page is up though, https://api.helioviewer.org/docs/v2

File: scripts/check_data_injection2.php [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+	$log = true;
+	
+	$output = shell_exec('ps -ef | grep python');
+	
+	$logBody = "////////////////////\n".date("Y-m-d H:i:s")."\n";
+	$logBody .= $output."";
+	//$strOut = array();
+	putenv("http_proxy=http://gs600-squid.ndc.nasa.gov:443");
+	putenv("https_proxy=http://gs600-squid.ndc.nasa.gov:443");
+	putenv("http_export=http://gs600-squid.ndc.nasa.gov:443");
```
[the remark this answers is not in the document]
Comment (dgarciabriseno) [reply scripts/check_data_injection2.php:25]: Got it, will move them over to one.

File: scripts/hv_stats/generate_bokeh_plots.sh [hunk 13]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (wafels) [inline scripts/hv_stats/generate_bokeh_plots.sh:1]: So this file runs a python script that autogenerates the content for statistics/bokeh and then moves it into the web directory. I think everything in statistics/bokeh should not be in the main repo, so remove them from the PR.

**Comment by wafels** [comment]
All log and csv files should not be in the repo. Most of them are old.

**Comment by wafels** [comment]
Also nohup.out

File: scripts/hv_stats/Jhv_movies/histogram.html (hunk 9 above)
Comment (wafels) [reply scripts/hv_stats/Jhv_movies/histogram.html:1]: Yes, it's autogenerated by hv_stats/hv_stats.py. It queries our databases and uses the python package bokeh to autogenerate plots. The subdirectories coverages, embed_service, histograms, hv_movies,hv_screenshots, hv_student, Jhv_movies and service_usage are all autogenerated and should not be part of the repo. The script could stand to be a little better, for example, the log files should go in a proper log directory. I'd hope there is a better solution for such autogenerated content.

**Comment by wafels** [comment]
Checking on the server, I think almost everything in docroot/statistics/bokeh is a copy of what hv_stats.py generates. Only index.html and style.css should be kept in docroot/statistics/bokeh . The file service_usage_fraction.html has not been updated since last year, so we must have dropped support for that. That file can be deleted too.

File: scripts/check_data_injection2.php (hunk 12 above)
[the remark this answers is not in the document]
Comment (dgarciabriseno) [reply scripts/check_data_injection2.php]: They're in a config file now, check the latest code

File: scripts/check_data_injection.php (hunk 11 above)
Comment (dgarciabriseno) [reply scripts/check_data_injection.php]: They're in a config file now, check the latest code

File: scripts/hv_stats/Jhv_movies/histogram.html (hunk 9 above)
Comment (dgarciabriseno) [reply scripts/hv_stats/Jhv_movies/histogram.html:1]: I think it's okay that they're here. Auto-generated files should be in .gitignore though so I'll add them to that.

File: scripts/hv_stats/generate_bokeh_plots.sh (hunk 13 above)
Comment (dgarciabriseno) [reply scripts/hv_stats/generate_bokeh_plots.sh:1]: Ok will remove.

**Comment by dgarciabriseno** [comment]
> All log and csv files should not be in the repo. Most of them are old.

> Also nohup.out

> Checking on the server, I think almost everything in docroot/statistics/bokeh is a copy of what hv_stats.py generates. Only index.html and style.css should be kept in docroot/statistics/bokeh . The file service_usage_fraction.html has not been updated since last year, so we must have dropped support for that. That file can be deleted too.

Updated all of these, check the latest.

**Comment by wafels** [comment]
scripts/hv_stats/hv_stats.tar.gz should be removed.

**Comment by wafels** [comment]
scripts/hv_stats/service_usage_fraction.html should be removed.

File: docroot/statistics/bokeh/Jhv_movies/histogram.html (hunk 6 above)
Comment (dgarciabriseno) [reply docroot/statistics/bokeh/Jhv_movies/histogram.html]: Addressed in a conversation below.

**Comment by dgarciabriseno** [comment]
> scripts/hv_stats/hv_stats.tar.gz should be removed.

> scripts/hv_stats/service_usage_fraction.html should be removed.

Removed.

File: src/Database/Statistics_backup.php [hunk 14]
```diff
@@ -0,0 +1,1763 @@
+<?php
+/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */
```
Comment (dgarciabriseno) [inline src/Database/Statistics_backup.php:2]: Remove "backup" file

File: src/Image/JPEG2000/JPXImage.new.php [hunk 15]
```diff
@@ -0,0 +1,141 @@
+<?php
+/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */
```
Comment (dgarciabriseno) [inline src/Image/JPEG2000/JPXImage.new.php:2]: Remove new/old file. Keep the main one.

**Pull request merged**